pkg/apis/pingcap/v1alpha1/backup.go

```diff
@@ -187,3 +222,77 @@ func IsCleanCandidate(backup *Backup) bool {
 func NeedNotClean(backup *Backup) bool {
 	return backup.Spec.CleanPolicy == CleanPolicyTypeOnFailure && !IsBackupFailed(backup)
 }
+
+// ParseLogBackupSubCommand parse the log backup subcommand from cr.
+// The parse priority of the command is stop > truncate > start.
+func ParseLogBackupSubcommand(backup *Backup) LogSubCommandType {
+	if backup.Spec.Mode != BackupModeLog {
+		return ""
+	}
+	if backup.Spec.LogStop {
+		return LogStopCommand
+	}
+	if backup.Spec.LogTruncateUntil != "" {
+		return LogTruncateCommand
+	}
+	return LogStartCommand
+}
+
+// IsLogBackupSubCommandOntheCondition return whether the log subcommand on the condition.
+func IsLogBackupSubCommandOntheCondition(backup *Backup, conditionType BackupConditionType) bool {
+	command := ParseLogBackupSubcommand(backup)
+	switch command {
+	case LogStartCommand, LogStopCommand:
+		if subStatus, ok := backup.Status.LogSubCommandStatuses[command]; ok {
+			return subStatus.Phase == conditionType
+		}
+	case LogTruncateCommand:
+		// truncate Command's truncating until is the spec truncate until means the truncate is in progress.
+		if subStatus, ok := backup.Status.LogSubCommandStatuses[command]; ok {
+			return subStatus.LogTruncatingUntil == backup.Spec.LogTruncateUntil && subStatus.Phase == conditionType
+		}
+	default:
+		return false
+	}
+	return false
+}
+
+// GetLogSumcommandConditionInfo gets log subcommand current phase's reason and message
+func GetLogSumcommandConditionInfo(backup *Backup) (reason, message string) {
+	command := ParseLogBackupSubcommand(backup)
+	if subStatus, ok := backup.Status.LogSubCommandStatuses[command]; ok {
+		for _, condition := range subStatus.Conditions {
+			if subStatus.Phase == condition.Type {
+				reason = condition.Reason
+				message = condition.Message
+				break
+			}
+		}
+	}
+	return
+}
+
+// IsLogBackupAlreadyStart return whether log backup has already started.
+func IsLogBackupAlreadyStart(backup *Backup) bool {
+	return backup.Spec.Mode == BackupModeLog && backup.Status.CommitTs != ""
+}
+
+// IsLogBackupAlreadyTruncate return whether log backup has already truncated.
+func IsLogBackupAlreadyTruncate(backup *Backup) bool {
+	if backup.Spec.Mode != BackupModeLog {
+		return false
+	}
+	// spec truncate Until TS <= start commit TS or success truncate until means log backup has been truncated.
+	var specTS, successedTS, startCommitTS uint64
+
+	specTS, _ = config.ParseTSString(backup.Spec.LogTruncateUntil)
```

Comment:
I think we shouldn't ignore the returned error. Different ts error may need return different result.

Reply:
`backup.Spec.LogTruncateUntil` will be checked in `validateBackup` which is in sync, so here can return false if there has an error, mean that just skip and let sync handle.